server/methods/accounts/accounts.js

```diff
@@ -1058,7 +1092,10 @@ export function setProfileCurrency(currencyName) {
   check(currencyName, String);
   if (this.userId) {
     Accounts.update(this.userId, { $set: { "profile.currency": currencyName } });
-    Hooks.Events.run("afterAccountsUpdate", this.userId, this.userId);
+    Hooks.Events.run("afterAccountsUpdate", this.userId, {
+      accountId: this.userId,
+      updatedFields: ["profile.currency"]
```

Comment:
Here you are passing in "profile.currency" as the field, but in `buildAccountSearchRecord` you are looking for fields like "firstName" rather than "profile.firstName", and for the address update, you're passing in the address fields directly without `profile.addressBook` prefix. So there needs to be consistency about whether you are passing in the full object path or just the last piece.

To simplify, you could just pass in the top-level field that changed, i.e. "profile". The downside is that you may then run the AccountSearch stuff more than you need to, but it would be easier to properly build the `updatedFields` array. I'm fine with whatever as long as it works and is consistent.

Reply:
I thought about passing the `profile` down, but as you mentioned, that will cause the it to run more frequently. All the addressBook is inside the `profile`, so it would actually run on every update, and would essentially be no different than what we are currently doing.

I'm going to go with just the last piece.